pages/[resourceType]/create.tsx

```diff
@@ -49,6 +49,77 @@ const CreateResourcePage = () => {
       </Center>
     </div>
   );
+
+  //handles what will happen when the submit button is clicked.
+  async function editorSubmitHandler() {
+    let customMessage: NotificationProps["message"] = <div>Problem connecting to server</div>;
+    let notifProps: NotificationProps = {
+      message: customMessage,
+      color: "red",
+      icon: <X size={18} />,
+      autoClose: false,
+    };
+
+    await fetch(`${process.env.NEXT_PUBLIC_DEQM_SERVER}/${resourceType}`, {
+      method: "POST",
+      body: codeEditorContents,
+      headers: {
+        "Content-Type": "application/json+fhir",
+      },
+    })
+      .then((response) => {
+        let newID;
+
+        if (response.status === 201 || response.status === 200) {
+          const regexResponse = NEW_ID_IN_HEADER_REGEX.exec(
+            response.headers.get("Location") as string,
+          ) as RegExpExecArray;
```

Comment:
I'm assuming this type cast is needed because theoretically the regexResponse could come back as null. Instead of casting it, could we have a case for `if (regexResponse == null)` and if it is just throw an error that the id couldn't be parsed off of the location header? Maybe in that case, we also say in the error message what the location header we got back is?

If an error is thrown in this case, TypeScript will then allow you to safely access regexResponse[0] below since it now knows the code can only have continued if it wasn't null